src/service/resources-impl.js

```diff
@@ -387,15 +388,40 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
+        // Try to immediately build element, it may already be ready.
+        resource.build();
+        this.schedulePass();
+      } else {
+        this.pendingBuildResources_.push(element);
+        this.buildReadyResources_();
+      }
     }
 
     dev.fine(TAG_, 'element added:', resource.debugid);
   }
 
   /**
+   * Builds resources that are ready to be built.
+   * @private
+   */
+  buildReadyResources_() {
```

Comment:
Or or, we can use binary search (What?! This may be the first time I've ever used a search algorithm in real life)! Find the first non-parse element, everything before is parse, everything after isn't.